tests/parsing.rs

```diff
@@ -0,0 +1,265 @@
+    assert_eq!(metadata.get_name().unwrap(), "rpm-feature-coverage");
+    assert_eq!(metadata.get_epoch().unwrap(), 1);
+    assert_eq!(metadata.get_version().unwrap(), "2.3.4");
+    assert_eq!(metadata.get_release().unwrap(), "5.el8");
+    assert_eq!(metadata.get_arch().unwrap(), "x86_64");
+    assert_eq!(
+        metadata.get_description().unwrap(),
+        "This package attempts to exercise many different features of RPM packages."
+    );
+    assert_eq!(
+        metadata.get_summary().unwrap(),
+        "A package for exercising many different features of RPM"
+    );
+    assert_eq!(metadata.get_license().unwrap(), "MPLv2");
+    assert_eq!(metadata.get_url().unwrap(), "http://bobloblaw.com");
+    assert_eq!(metadata.get_packager().unwrap(), "Michael Bluth");
+    assert_eq!(metadata.get_vendor().unwrap(), "Bluth Company");
+    assert_eq!(metadata.get_group().unwrap(), "Development/Tools");
+    assert_eq!(metadata.get_vcs().unwrap(), "https://github.com/rpm-rs/rpm");
+    assert_eq!(metadata.get_cookie().unwrap(), "localhost 1681068559");
+
+    assert_eq!(metadata.get_build_host().unwrap(), "localhost");
+    assert_eq!(metadata.get_build_time().unwrap(), 1681068559);
+
+    assert_eq!(
+        metadata.get_file_paths().unwrap(),
+        vec![
+            PathBuf::from("/etc/complex/pkg.cfg"),
+            PathBuf::from("/usr/bin/complex_a"),
+            PathBuf::from("/usr/share/doc/rpm-feature-coverage"),
+            PathBuf::from("/usr/share/doc/rpm-feature-coverage/README"),
+            PathBuf::from("/var/lib/complex"),
+            PathBuf::from("/var/log/complex.log"),
+        ]
+    );
+    assert_eq!(
+        metadata.get_file_checksums().unwrap(),
+        vec![
+            "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+            "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+            "",
+            "bc1c6832a5a5f002a2fd7435b58ce6852408b7750f9c1c4a69098972f44dd1b7",
+            "",
+            "",
+        ]
+    );
+
+    assert_eq!(
+        metadata.get_changelog_entries().unwrap(),
+        vec![
+            ChangelogEntry {
+                author: "George Bluth <george@federalprison.gov> - 3.3.3-3".to_owned(),
+                timestamp: 1623672000,
+                description: "- There’s always money in the banana stand".to_owned()
+            },
+            ChangelogEntry {
+                author: "Job Bluth <job@alliance-of-magicians.com> - 2.2.2-2".to_owned(),
+                timestamp: 1619352000,
+                description: "- I've made a huge mistake".to_owned()
+            },
+            ChangelogEntry {
+                author: "Lucille Bluth <lucille@bluthcompany.com> - 1.1.1-1".to_owned(),
+                timestamp: 1617192000,
+                description: "- It's a banana, Michael. How much could it cost, $10?".to_owned()
+            },
+        ]
+    );
+
+    // assert_eq!(
```

Comment:
TODO: need to finish https://github.com/rpm-rs/rpm/pull/69 before these are really practical to write tests for that use any kind of normal API.